### ISSUE TITLE
Define settings in sysbuild Kconfig and and propagate those to image(s) Kconfig

### ISSUE THREAD

**Issue by tejlmand** [zephyrproject-rtos/zephyr#53126]
Part of: #41640 

When using sysbuild to control multiple images, then it is often necessary to share a setting across builds.
And if a setting is shared, then ensure if a user reconfigures the setting, then this information is updated across images.

It is therefore required that it will be possible to define settings in sysbuild Kconfig and and then propagate those settings to image(s) Kconfig.
This should result in those settings becoming inaccessible by users, i.e. they should be promptless.

When implementing this, then the following must be considered:
- A setting must be shared to all images being build for same board / SoC (and same core on the SoC).
- A setting must be shared to all images being build for same board / SoC but targetting different cores on the SoC.
- A setting must go to all images regardless of the board used.

**Comment by nordicjm** [comment]
@de-nordic this seems like a good candidate for sharing information in builds between mcuboot and target images which can then be used to check if e.g. an update is too large to fit for a given mcuboot image.

**Comment by de-nordic** [comment]
> @de-nordic this seems like a good candidate for sharing information in builds between mcuboot and target images which can then be used to check if e.g. an update is too large to fit for a given mcuboot image.

Yes, this could allow us to better integrate builds of MCUmgr enabled apps with MCUboot and impose the MCUboot configuration on the former.
Maybe this can also help us with building apps for Direct XIP?
